### ISSUE TITLE
question about dimension handling

### ISSUE THREAD

**Issue by hw-kim-ivan** [holmdk/Video-Prediction-using-PyTorch#6]
hello, I came from your blog:) - https://holmdk.github.io/2020/04/02/video_prediction.html

![image](https://user-images.githubusercontent.com/71258448/120196670-11b7c200-c25b-11eb-9785-194fafcb9bf6.png)

Can we say input sequence is encoded through encoder even though dimension of last hidden state (12,10,64,64,64) of encoder is bigger than dimension of input sequence (12,10,1,64,64)?

**Comment by holmdk** [comment]
Hi. Yes, absolutely. The last hidden state from ConvLSTM Encoder 2 is a (temporal) representation of all previous hidden states from earlier layers. So, even though we do not reduce the width and height (in principle we could, however), this final hidden state still contains an encoding of all previous information. The idea is the same as a pure seq2seq LSTM encoder-decoder.

**Comment by hw-kim-ivan** [comment]
Thanks for kind explaination:) I literally understood that "encoder" compress the input data size into smaller representation, just like [12,1,32,32,32] = Encoder([12,10,1,64,64]).

**Comment by hw-kim-ivan** [comment]
I have another question abount normalization! what do you recommand for normalization between Layernorm or Batchnorm between layers in Encoder and Decoder? (or Groupnorm, instancenorm, etc...)

It's quite hard to find other implementations in using convlstm and pytorch lightning...
